### DISCUSSION TITLE
NPG and DAPG algorithm implementations

### DISCUSSION THREAD

**Discussion by alessandroassirelli98** [Toni-SM/skrl#168] [Ideas]
Hello everybody and thanks for the great repository!
I am currently working on reinforcement learning and reinforcement learning from demo.
I see there is no implementation for Natural Policy Gradient. I was thinking to implement it. Do you think it's a good idea? specifically in its Demonstration Augmented version (DAPG) which seems to be very powerful. I know this repo is mainly for pure RL, but maybe it can be extended

Thanks and let me know!

**Comment by Toni-SM** [comment]
Hi @alessandroassirelli98 

Implementing new algorithms is always welcome :)

In this case, my proposal to you is to implement first the NPG version and then, as an independent algorithm (in a separate folder), the DAPG algorithm without crossing them (even in there is duplicated code). This, with the objective of keeping the code clean and  with no other degree of dependency than the base class.
